datafusion/datasource/src/mod.rs

```diff
@@ -313,6 +314,78 @@ async fn find_first_newline(
     Ok(index)
 }
 
+/// Generates test files with min-max statistics in different overlap patterns
+/// Used by tests and benchmarks
+pub fn generate_test_files(num_files: usize, overlap_factor: f64) -> Vec<FileGroup> {
```

Comment:
Also, based on my recent experience working in statistics, this function will be very helpful for testing functionality later